### PR TITLE
Features/feature set

### PR DESCRIPTION
**Pull request by sharathrnair87**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] Where applicable I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

* `tfectl run` now shows the duration of the run for a subset of run types, currently supports `planned_and_finished`, `applied`, `planned_and_saved` and `policy_checked`
* `tfectl workspace` returns the average duration of a plan or plan-and-apply run for a given workspace.
* Updated test suite to include `run.go`
* The underlying mechanism to print the JSON output was changed from `fmt.Println` to `cmd.Println`, in order to capture the output of the Cobra commands for the unit tests. The former results in the outputs leaking out to STDOUT and STDERR and therefore not captured in the user-defined output buffesr used to verify outputs in the tests.
* The `JqRun` method now returns `[]byte` and `error` types as opposed to printing the output itself

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?
```bash
==> Testing <==
cd cmd/ && go test -v
=== RUN   TestPolicySetListCmd
--- PASS: TestPolicySetListCmd (1.20s)
=== RUN   TestPolicyListCmd
--- PASS: TestPolicyListCmd (1.16s)
=== RUN   TestRegistryModulesListCmd
--- PASS: TestRegistryModulesListCmd (11.00s)
=== RUN   TestRootCmd
--- PASS: TestRootCmd (0.00s)
=== RUN   TestRunListCmd
--- PASS: TestRunListCmd (15.58s)
=== RUN   TestTagsListCmd
--- PASS: TestTagsListCmd (3.52s)
=== RUN   TestTeamListCmd
--- PASS: TestTeamListCmd (6.14s)
=== RUN   TestWorkspaceListCmd
--- PASS: TestWorkspaceListCmd (24.62s)
PASS
ok      github.com/AGLEnergyPublic/tfectl/cmd   63.226s
```

* Runs
```bash
./tfectl run list --workspace-id "ws-dmGfXPBqrYMuoNYV" --status "planned_and_finished"
[
  {
    "id": "run-RXoAsLT6GNF63vxz",
    "workspace_id": "ws-dmGfXPBqrYMuoNYV",
    "workspace_name": "wrkspace-test-1",
    "status": "planned_and_finished",
    "run_duration": "90.962000"
  },
  {
    "id": "run-pxzpjjc45UEpi6oe",
    "workspace_id": "ws-dmGfXPBqrYMuoNYV",
    "workspace_name": "wrkspace-test-2",
    "status": "planned_and_finished",
    "run_duration": "92.367000"
  }
]
```

* Workspace
```bash
./tfectl workspace get --ids ws-dmGfXPBqrYMuoNYV
[
  {
    "name": "uat-injest",
    "id": "ws-dmGfXPBqrYMuoNYV",
    "locked": false,
    "execution_mode": "agent",
    "terraform_version": "1.7.0",
    "tags": [
      "env:uat"
      "application:injester",
    ],
    "created_days_ago": "408.213596",
    "updated_days_ago": "296.819558",
    "last_remote_run_days_ago": "296.822385",
    "last_state_update_days_ago": "296.819586",
    "average_run_duration": "162.537500"
  }
]
```

### CODE REVIEW

File: cmd/policy_set.go [hunk 1]
```diff
@@ -91,7 +91,30 @@
...
+      "projects":%v,
+      "project_count":%d,
+      "policies":%v,
+      "policy_count":%d
+    }`,
+				policySet.Name,
+				policySet.ID,
+				policySet.Kind,
+				policySet.Global,
+				string(workspaceSlice),
```
[the remark this answers is not in the document]
Comment (sharathrnair87) [reply cmd/policy_set.go:111]: CodeQL false positive - https://github.com/github/codeql/issues/14159

File: cmd/policy_set.go [hunk 2]
```diff
@@ -91,7 +91,30 @@
...
+				policySet.WorkspaceCount,
+				string(workspaceExclSlice),
```
[the remark this answers is not in the document]
Comment (sharathrnair87) [reply cmd/policy_set.go:113]: CodeQL false positive - https://github.com/github/codeql/issues/14159

File: cmd/policy_set.go [hunk 3]
```diff
@@ -91,7 +91,30 @@
...
+				string(projectSlice),
```
[the remark this answers is not in the document]
Comment (sharathrnair87) [reply cmd/policy_set.go:114]: CodeQL false positive - https://github.com/github/codeql/issues/14159

File: cmd/policy_set.go [hunk 4]
```diff
@@ -91,7 +91,30 @@
...
+				policySet.ProjectCount,
+				string(policiesSlice),
```
[the remark this answers is not in the document]
Comment (sharathrnair87) [reply cmd/policy_set.go:116]: CodeQL false positive - https://github.com/github/codeql/issues/14159

**Review by a30004053 — APPROVED** [review]

**Review by a30000931 — APPROVED** [review]

**Pull request merged**